### PR TITLE
GCP VPC

### PR DESCRIPTION
**Pull request by quasiben**
resolves https://github.com/dask/dask-cloudprovider/issues/215

PR allow users to define the VPC network to use when launching clusters.  For example, building a network with the following 
firewall rules restricts accessing the machines to 22, 8786, and 8787 and allows all machines in the VPC to access one another and any external address (the internet)

Screen Shot 2020-12-14 at 12 29 25 PM

cc @eric-czech

### CODE REVIEW

**Comment by eric-czech** [comment]
Awesome, thanks @quasiben!

File: dask_cloudprovider/gcp/instances.py [hunk 1]
```diff
@@ -536,6 +542,7 @@ def __init__(
             "zone": zone or self.config.get("zone"),
             "machine_type": self.machine_type,
             "ngpus": ngpus or self.config.get("ngpus"),
+            "network": network or self.config.get("network"),
```
Comment (jacobtomlinson) [inline dask_cloudprovider/gcp/instances.py:545]: I think we also need to add the default to the `cloudprovider.yaml` file.

File: dask_cloudprovider/gcp/instances.py [hunk 2]
```diff
@@ -111,7 +114,7 @@ def create_gcp_config(self):
             "networkInterfaces": [
                 {
                     "kind": "compute#networkInterface",
-                    "subnetwork": f"projects/{self.projectid}/regions/{self.general_zone}/subnetworks/default",
+                    "subnetwork": f"projects/{self.projectid}/regions/{self.general_zone}/subnetworks/{self.network or 'default'}",
```
Comment (jacobtomlinson) [inline dask_cloudprovider/gcp/instances.py:117]: This default value should probably live in the `cloudprovider.yaml` file.

**Review by jacobtomlinson — COMMENTED** [review]
Thanks @quasiben for looking at this. Couple of comments.

**Comment by jacobtomlinson** [comment]
Also can we add a section to the docs detaiing the rules you shared above?

**Review by jacobtomlinson — APPROVED** [review]
Awesome thanks!

**Pull request merged**